ui/core/components/suggest_reforges_action.tsx

```diff
@@ -859,6 +985,12 @@ export class ReforgeOptimizer {
 			this.setPseudoStatCoefficient(coefficients, PseudoStat.PseudoStatSpellHitPercent, appliedAmount);
 		}
 
+		// If a highest Stat constraint is to be enforced, then update the associated
+		// coefficient if applicable.
+		if (this.relativeStatCap) {
+			this.relativeStatCap!.updateCoefficients(coefficients, stat, amount);
```

Comment:
You could just do: `this.relativeStatCap?.updateCoefficients(coefficients, stat, amount);` so it will only execute when it's defined. Otherwise the `!` is not needed since you null check in the if.